<launch>
  <!-- Launch an example that sends commands via C++ API. -->

  <arg name="camera_frame_name" default="ee_link" />
  <arg name="gravity_frame_name" default="base_link" />
  <arg name="target_frame_name" default="demo_frame" />
  <arg name="loop_rate" default="50" /> <!-- hz -->
  <arg name="rotational_tolerance" default="0.05" /> <!-- radians -->

  <arg name="look_at_pose_server_name" default="/look_at_pose" />
  <arg name="target_pose_publish_topic" default="target_pose" />

  <!-- SERVICE CALL TO SPECIFY START AND STOP CONTROLLERS -->
  <node pkg="rosservice" type="rosservice" name="controller_manager_service"
        args="call --wait /controller_manager/switch_controller 
              '{ start_controllers: ['joint_group_position_controller'],
                 stop_controllers: ['arm_controller'],
                 strictness: 2,
                 start_asap: True,
                 timeout: 0 }' " />

  <!-- NODE TO CALCULATE TARGET POSE IN CAMERA FRAME -->
  <node name="look_at_pose_server" pkg="look_at_pose" type="look_at_pose_server" output="screen" />

  <!-- PUBLISH A TARGET FRAME FOR DEMONSTRATION -->
  <node name="target_frame_pub" pkg="tf" type="static_transform_publisher" args="1 1 1 0 0 0 /world /demo_frame 100" />

  <!-- NODE CONTAINING SERVO AND THE CAMERA POINTER COMMAND -->
  <node name="servo_camera_pointer" pkg="servo_camera_pointer" type="camera_pointer" output="screen">
    <rosparam command="load" file="$(find servo_camera_pointer)/config/ur_simulated_config.yaml" />
    <rosparam command="load" file="$(find moveit_servo)/config/pose_tracking_settings.yaml" />

    <param name="camera_frame_name" type="str" value="$(arg camera_frame_name)" />
    <param name="gravity_frame_name" type="str" value="$(arg gravity_frame_name)" />
    <param name="target_frame_name" type="str" value="$(arg target_frame_name)" />
    <param name="loop_rate" type="double" value="$(arg loop_rate)" />
    <param name="rotational_tolerance" type="double" value="$(arg rotational_tolerance)" />
    <param name="look_at_pose_server_name" type="str" value="$(arg look_at_pose_server_name)" />
    <param name="target_pose_publish_topic" type="str" value="$(arg target_pose_publish_topic)" />
  </node>

</launch>
